GithubApiWithVIPER/GithubApiWithVIPER/API/ApiUrls.swift

```diff
@@ -12,21 +12,51 @@ struct Constant {
     static let baseUrl = "https://api.github.com/"
 }
 
-enum APIRouter {
-    case searchRepository(keyword: String)
+protocol ApiRequestProtocol {
+    var method: HttpMethod { get }
+    var url: URL! { get }
+    var body: Encodable? { get }
+}
+
+struct SearchRepoRequest: ApiRequestProtocol {
+    
+    var url: URL!
+    var method: HttpMethod = .GET
+    var body: Encodable? = nil
     
-    var url: String {
-        switch self {
-        case .searchRepository(let keyword):
-            return Constant.baseUrl + "search/repositories?q=\(keyword)"
+    mutating func generateUrl(keyword: String?) {
```

Comment:
Bu yöntem çok iyi değil. Şimdi bir ApiRequestProtocol'ü url'siz olabilir mi? olamaz o yüzden bunu struct'in init'inde alıp orada url'i oluşturacaksın. Ayrıca bu sayede URL! kullanmana da gerek kalmaz. 

1. generateUrl'i silip keyword'ü struct init'inde almaca
2. fatalError'u dikkatli kullanmak lazım uygulamayı crash ettiği için. Burada safe çünkü baseUrl + search path'i düzgün ve patlama ihtimali düşük hatta yok gibi.

Reply:
Haklısın abi dediğin şekilde yapmak daha doğru.